### PR TITLE
update native_filters behavior for GCR 0.7.1

### PR DESCRIPTION
**Pull request by yymao**
In GCR 0.7.x, one can use `native_filters.check_scalar` to check if a dict of scalar satisfies `native_filters`. This makes the code that checks `native_filters` much more readable. 

This also enables the users to write `native_filters` in string, for example: `native_filters=['tract == 4850']`. 

Note that in order to test this, you need GCR 0.7.1. 

cc @EiffL @danielsf

### CODE REVIEW

**Comment by danielsf** [comment]
Running this branch on NERSC's Jupyter dev, I observed that this command
```
x_cat = GCRCatalogs.load_catalog('proto-dc2_v2.1.2')
x_q = x_cat.get_quantities(['ra', 'dec', 'galaxy_id'])
```
resulted in
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-97-710a42ff9dc6> in <module>()
----> 1 x_q = x_cat.get_quantities(['ra', 'dec', 'galaxy_id'])

/global/u1/d/danielsf/generic-catalog-reader-yymao/GCR.py in get_quantities(self, quantities, filters, native_filters, return_iterator)
    121 
    122         data_all = defaultdict(list)
--> 123         for data in it:
    124             for q in quantities:
    125                 data_all[q].append(data[q])

/global/u1/d/danielsf/generic-catalog-reader-yymao/GCR.py in _get_quantities_iter(self, quantities, filters, native_filters)
    504 
    505     def _get_quantities_iter(self, quantities, filters, native_filters):
--> 506         for native_quantity_getter in self._iter_native_dataset(native_filters):
    507             data = self._load_quantities(quantities.union(set(filters.variable_names)),
    508                                          native_quantity_getter)

/global/u1/d/danielsf/gcr-catalogs-desc/GCRCatalogs/alphaq.py in _iter_native_dataset(self, native_filters)
    351     def _iter_native_dataset(self, native_filters=None):
    352         if native_filters is not None:
--> 353             raise ValueError('*native_filters* is not supported')
    354         with h5py.File(self._file, 'r') as fh:
    355             def _native_quantity_getter(native_quantity):

ValueError: *native_filters* is not supported
```

I was running `master` of `generic-catalog-reader`.  Maybe something else about my environment was broken.  We should check for this, though.

**Comment by yymao** [comment]
@danielsf, thanks! Good catch. There was an API inconsistency. Fixed now in GCR 0.7.1.

**Comment by EiffL** [comment]
In the middle of testing this for dc2, I get an error when not using native filters:

```
>>> data = catalog.get_quantities(['ra', 'dec'])
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-17-8e6406ea7d7a> in <module>()
----> 1 data = catalog.get_quantities(['ra', 'dec'])

~/repo/generic-catalog-reader/GCR.py in get_quantities(self, quantities, filters, native_filters, return_iterator)
    124 
    125         data_all = defaultdict(list)
--> 126         for data in it:
    127             for q in quantities:
    128                 data_all[q].append(data[q])

~/repo/generic-catalog-reader/GCR.py in _get_quantities_iter(self, quantities, filters, native_filters)
    512 
    513     def _get_quantities_iter(self, quantities, filters, native_filters):
--> 514         for native_quantity_getter in self._iter_native_dataset(native_filters):
    515             data = self._load_quantities(quantities.union(set(filters.variable_names)),
    516                                          native_quantity_getter)

~/repo/gcr-catalogs/GCRCatalogs/dc2_coadd.py in _iter_native_dataset(self, native_filters)
    375         for dataset in self._datasets:
    376             dataset_info = self.get_dataset_info(dataset)
--> 377             if not native_filters.check_scalar(dataset_info):
    378                 continue
    379 

AttributeError: 'NoneType' object has no attribute 'check_scalar'
```

**Comment by EiffL** [comment]
Looks like it should be caught in the coadd reader

File: GCRCatalogs/buzzard.py [hunk 1]
```diff
@@ -224,12 +224,8 @@ def _generate_native_quantity_list(self):
 
     def _iter_native_dataset(self, native_filters=None):
         for healpix in self.healpix_pixels:
-
-            fargs = dict(healpix_pixel=healpix)
-            if native_filters and not all(f[0](*(fargs[k] for k in f[1:])) for f in native_filters):
-                continue
-
-            yield functools.partial(self._native_quantity_getter, healpix=healpix)
+            if native_filters.check_scalar({'healpix_pixel': healpix}):
```
Comment (EiffL) [inline GCRCatalogs/buzzard.py]: need to check that native_filters is not none

File: GCRCatalogs/dc2_coadd.py [hunk 2]
```diff
@@ -374,10 +374,7 @@ def _generate_native_quantity_list(self):
     def _iter_native_dataset(self, native_filters=None):
         for dataset in self._datasets:
             dataset_info = self.get_dataset_info(dataset)
-            if native_filters and \
-                    not all(native_filter[0](
-                        *(dataset_info[c] for c in native_filter[1:])) \
-                            for native_filter in native_filters):
+            if not native_filters.check_scalar(dataset_info):
```
Comment (EiffL) [inline GCRCatalogs/dc2_coadd.py]: need to check that filters is not none

**Comment by yymao** [comment]
Good catch, thanks @EiffL

**Comment by EiffL** [comment]
hahaha, you're fixing the code faster than I review it

**Pull request merged**